vmm/src/serial_manager.rs

```diff
@@ -330,20 +329,20 @@ impl SerialManager {
                                         listener.accept().map_err(Error::AcceptConnection)?;
                                     let writer =
                                         unix_stream.try_clone().map_err(Error::CloneUnixStream)?;
-                                    reader = Some(
-                                        unix_stream.try_clone().map_err(Error::CloneUnixStream)?,
-                                    );
-
-                                    epoll::ctl(
-                                        epoll_fd,
-                                        epoll::ControlOptions::EPOLL_CTL_ADD,
-                                        unix_stream.into_raw_fd(),
-                                        epoll::Event::new(
-                                            epoll::Events::EPOLLIN,
-                                            EpollDispatch::File as u64,
-                                        ),
-                                    )
-                                    .map_err(Error::Epoll)?;
+                                    reader = Some(unix_stream);
+                                    if let Some(ref stream) = reader {
```

Comment:
What's the point of wrapping it in a `Some` if you're then going to access the internal again immediately?

Reply:
`reader` is defined as
```
let mut reader: Option<UnixStream> = None;
```